File: python/grains/grains.py
""" Calculate the number of grains of wheat on a chessboard 
    given that the number on each square doubles.
"""
GRAINS = list(range(64))

def square(number: int) -> int or ValueError:
    """ Return number of grains given the square number """
    if number < 1 or number > 64:
        raise ValueError('square must be between 1 and 64')

    for idx, _ in enumerate(GRAINS):
        if idx == 0:
            GRAINS[idx] = 1
            continue
        GRAINS[idx] = GRAINS[idx - 1] * 2

    return GRAINS[number - 1]

def total() -> int:
    """ Return total number of grains on the board """
    return sum(GRAINS)

File: go/grains/grains.go
package grains

import (
	"errors"
	"math"
)

func Square(number int) (uint64, error) {
	grains := make([]uint64, 64)

	if number < 1 || number > 64 {
		return 0, errors.New("must be between 1 and 64, inclusive")
	}

	for idx := range grains {
		if idx == 0 {
			grains[idx] = 1
			continue
		}
		grains[idx] = grains[idx-1] * 2
	}

	return grains[number-1], nil
}

func Total() uint64 {
	return math.MaxUint64
}
